ui/src/widgets/ui-notification/UINotification.vue

```diff
@@ -5,21 +5,29 @@
         vertical
         multi-line
         :timeout="-1"
-        :location="props.position"
+        :location="position"
         :style="{'--nrdb-ui-notification-color': color}"
     >
-        <div v-if="props.showCountdown" class="nrdb-ui-notification-countdown">
-            <v-progress-linear v-model="countdown" :color="messages[id]?.color || (props.colorDefault ? 'primary' : color)" style="display: block; width: 100%" />
+        <div v-if="showCountdown" class="nrdb-ui-notification-countdown">
+            <v-progress-linear v-model="countdown" :color="color" style="display: block; width: 100%" />
```

Comment:
@bartbutenaers I noticed that the progress bar of the notification is not picking the fill colour even the `Show timeout countdown bar on notification
 ` is checked. And when I investigate I found that this change `:color="color"` has caused it. 
When the default colour option is checked in the NR it picks the group colour `rgb(var(--v-theme-group-background))` which is handled in [this line](https://github.com/FlowFuse/node-red-dashboard/pull/932/files#diff-ba8467bf1f890995ffb760b3d600a311b3d3ef2d09afc82407ee769f7c8fb01aR72:~:text=.colorDefault)%20%7B-,return,-%27rgb(var)

<img width="365" alt="Screenshot 2024-08-16 at 15 53 23" src="https://github.com/user-attachments/assets/31ff7f3a-d41d-4e19-991e-0906af9e879d">

As in the existing implementation group colour applies to the background of the notification widget as well.

Reply:
@gayanSandamal Thanks a lot for your review!

I really think that I had updated that code, but will have done something wrong in one or another commit.    Appreciate that you have shared your solution!!  Will change it tonight.

> As in the existing implementation group colour applies to the background of the notification widget as well.

Just to make sure: is this also solved using your fix?

Reply:
@bartbutenaers no worries at all with reviewing 🙌

It worked when I reverted and tried with the old code at line [12](https://github.com/FlowFuse/node-red-dashboard/pull/932/files/b0c10830ab60128686e41d02966aed7f4bfb13f5#diff-ba8467bf1f890995ffb760b3d600a311b3d3ef2d09afc82407ee769f7c8fb01a:~:text=%3Cv%2Dprogress%2Dlinear%20v%2Dmodel%3D%22countdown%22%20%3Acolor%3D%22messages%5Bid%5D%3F.color%20%7C%7C%20(props.colorDefault%20%3F%20%27primary%27%20%3A%20color)%22%20style%3D%22display%3A%20block%3B%20width%3A%20100%25%22%20/%3E)

Reply:
@gayanSandamal,
I had indeed not paid attention enough to the progress bar color.
But not sure how to handle it, so some help would be welcome...

The original color for the progress bar worked like this:
```
"messages[id]?.color || (props.colorDefault ? 'primary' : color)
```

So I need to introduce a new computed field `progressColor` with the same logic as the original one:

![image](https://github.com/user-attachments/assets/095505f0-1396-46b2-9c66-3d392cafbc20)

However recently @joepavitt has introduced (kudos!) a new way of working for dynamic properties.  So the color is not available anymore in the messages, which means I cannot use that statement anymore.  Not sure if I should get the dynamic color now from somewhere else.  Because in the 'else' part I will get the static or dynamic color via getProperty (via `this.color`).

I am not sure which behaviour you guys prefer, but perhaps you have something simple in your head?

Thanks!!!!

Reply:
You can achieve this by using the following code. I've moved the same existing logic to the `progressColor` computed property:

```
progressColor () {
    return this.messages[this.id]?.color || (this.props.colorDefault ? 'primary' : this.color)
},
```

<img width="850" alt="Screenshot 2024-08-18 at 20 37 51" src="https://github.com/user-attachments/assets/31319154-4ea9-467d-9a6c-529212d69dec">

@joepavitt What do you think regarding this?

Reply:
@gayanSandamal,
Thanks for thinking out loud!!  

BTW I sometimes ask questions during the weekend (because I have a bit more free time then), but that doesn't mean I expect you guys to respond during the weekends ;-)

> I've moved the same existing logic to the progressColor computed property

That was indeed my first thought, but I didn't even tried it because the theory in my brain tells me it wouldn't work.  But perhaps my theoretical knowledge is incorrect...

+ Recently Joe had redesigned the dynamic property mechanism.  Since then I expect the color to be provided via `msg.ui_update.color` (instead of `msg.color`), which will be stored in the *state store*.  
+ The messages are - from this PR on - only stored in the *data store* when a payload is provided.  Because the text (to be displayed) does not to be persisted in the state store, so it is stored (as `msg.payload`) in the data store.  Because if you would inject a message without payload (e.g. to set only the color), then the message would disappear.
+ So I don't expect that the color can be found anymore in the stored messages (in the data store).  Which means I don't expect the `this.messages[this.id]?.color` to be available anymore.
+ And in the new dynamic property mechanism, the `this.color` will - via `this.getProperty('color')` - will return the dynamic color value or the static property value.  So in fact the old `this.messages[this.id]?.color` is now returned by `this.color` (only if a dynamic property value has been injected).

This is the reason why I am doubting how the statement should be migrated...

Reply:
@bartbutenaers thanks for the clarification.

Regarding "this.messages[this.id]?.color" I will check this with @joepavitt offline and provide my feedback

Reply:
We are in that awakwars position where we haven't yet migrated over `ui-notification` to the new API - so as Bart has put it is fine. Don't want to overload this PR with the migration to the new API _as well_

Reply:
@joepavitt Thanks for the clarification.

@bartbutenaers Based on Joe's feedback, it makes sense to proceed with the current approach, sticking with msg.color for now.

Reply:
> Don't want to overload this PR with the migration to the new API as well

@joepavitt,
I had last week also updated this node (in this PR) to migrate it to the new API.  Thought that would be helpful for you guys, to get a bit of pressure from your shoulders.  Hopefully that is not a problem, because I don't really have much time to refactor it again to use the old API...

But that means that `msg.color` is not used anymore for this node, which is why I don't know how to migrate `this.messages[this.id]?.color`

Reply:
> this.messages[this.id]?.color

Would instead then just be covered by the new API's `getProperty`